File: apps/backend/app/schemas/notificacion.py
"""
Schemas Pydantic del modulo de notificaciones internas.
"""

from datetime import datetime

from pydantic import BaseModel

from app.core.constants import CanalNotificacion, EstadoNotificacion


class NotificacionItem(BaseModel):
    id: str
    incidente_id: str | None = None
    tipo_evento: str
    canal: CanalNotificacion
    estado: EstadoNotificacion
    asunto: str | None = None
    contenido: str
    fecha_envio: datetime | None = None
    fecha_lectura: datetime | None = None
    created_at: datetime


class NotificacionListResponse(BaseModel):
    items: list[NotificacionItem]
    total: int
    unread_count: int


class NotificacionUnreadCount(BaseModel):
    unread_count: int

File: packages/shared-types/src/notificacion.ts
/**
 * Contratos compartidos del modulo de notificaciones internas.
 */

import { CanalNotificacion, EstadoNotificacion } from "./enums";

export interface NotificacionItem {
  id: string;
  incidente_id: string | null;
  tipo_evento: string;
  canal: CanalNotificacion;
  estado: EstadoNotificacion;
  asunto: string | null;
  contenido: string;
  fecha_envio: string | null;
  fecha_lectura: string | null;
  created_at: string;
}

export interface NotificacionListResponse {
  items: NotificacionItem[];
  total: number;
  unread_count: number;
}

export interface NotificacionUnreadCount {
  unread_count: number;
}
